### PR TITLE
Implement Map Class

### PR DESCRIPTION
**Pull request by mcockle23**
## Description

Adds the Map Class for parsing input, removing whitespace, punctuation, and capitalization.

## Checklist

Please ensure that your pull request meets the following requirements:

- [ ] Code compiles without errors and warnings
- [ ] Code follows the project's coding style guidelines
- [ ] Code is tested and test cases pass
- [ ] Code has been reviewed by at least one other developer

## Related Issues

https://github.com/saie100/ood_map_reduce/issues/3

### CODE REVIEW

File: Map.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    allswell += "king's sake, he were living! I think it would be\n";
+    allswell += "the death of the king's disease.\n";
+    allswell += "LAFEU\n";
+    allswell += "How called you the man you speak of, madam?\n";
+    allswell += "COUNTESS\n";
+    allswell += "He was famous, sir, in his profession, and it was\n";
+    allswell += "his great right to be so: Gerard de Narbon.\n";
+
+    fileName = "all's well that end's well";
+    mapguy.map(fileName, allswell);
```
Comment (vnijat93) [inline Map.cpp:129]: qq - how does this unit test work? I don't see an `assert`

Comment (mcockle23) [reply Map.cpp:129]: It's not actually a unit test. I was just using the strings to test the parsing by displaying it on the console. I'm planning to set up smaller strings that test specific parts of the map function when I create the unit tests.

File: src/Map.cpp [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+// default constructor
+Map::Map() {}
+
+// checks if the passed in character is a punctuation or whitespace character
+bool isPunctOrSpace(int aChar) {
+    return (ispunct(aChar) || isspace(aChar));
+}
+
+// tokenizes the value into distinct words, removing whitespace, punctuation, and capitalization
+void Map::map(const string& aKey, const string& aValue) {
```
Comment (vnijat93) [inline src/Map.cpp:33]: can you have this method to accept a `filaPath` where the Map object will write the temp data to

Comment (mcockle23) [reply src/Map.cpp:33]: Yes, I will add that in a future revision when I add the FileManager calls.

**Review by vnijat93 — APPROVED** [review]
lgtm

**Pull request merged**